operating_systems.go

```diff
@@ -7,12 +7,12 @@ package user_agent
 import "strings"
 
 // Normalize the name of the operating system. By now, this just
-// affects to Windows.
+// affects to Windows NT.
 //
 // Returns a string containing the normalized name for the Operating System.
 func normalizeOS(name string) string {
 	sp := strings.SplitN(name, " ", 3)
-	if len(sp) != 3 {
+	if len(sp) != 3 || sp[1] != "NT" {
```

Comment:
Hey,

The reason I added it, is because "Windows Phone 10" will be matched as
"Windows 10" and not as "Windows Phone 10" as indicated in your tests.

The reason it was matched as "Windows phone 10" at first place is just
because it would not match "6.4" and therefore, name was returned at the
end of NormalizOS function.

In normailzeOS "case" you just normalize Nt versions, so "Windows phone 10"
for example, should be returned as is. otherwise it will be matched in
"10.0" case (relevant to Windows NT only) and "Windows 10" will be returned.

Another solution might be to send only NT versions to NormailzeOS, but
according to your comment above, it might be used for parsing other
(non-NT) Windows versions.

Thanks and sorry about all the mentions in issue #21, won't happen again.

Thanks,
Niv
On Dec 29, 2015 2:14 PM, "Miquel Sabaté Solà" <notifications@github.com>
wrote:

> In operating_systems.go
> <https://github.com/mssola/user_agent/pull/24#discussion_r48536655>:
>
> >  //
> >  // Returns a string containing the normalized name for the Operating System.
> >  func normalizeOS(name string) string {
> >  	sp := strings.SplitN(name, " ", 3)
> > -	if len(sp) != 3 {
> > +	if len(sp) != 3 || sp[1] != "NT" {
>
> Is this really necessary ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mssola/user_agent/pull/24/files#r48536655>.
>